commit a4abf22c79b517e989a50a21c5a80e72714aad53
Author: Marco Martin <notmart@gmail.com>
Date:   Wed Apr 20 18:10:02 2011 +0200

    fix popup position
    
    backport:
    if item is in an hidden Dialog, try to find the parent applet and use it instead of item
    this fixes the device notifier popup placement

diff --git a/plasma/corona.cpp b/plasma/corona.cpp
index 9469428..c6fedb1 100644
--- a/plasma/corona.cpp
+++ b/plasma/corona.cpp
@@ -747,6 +747,31 @@ QPoint Corona::popupPosition(const QGraphicsItem *item, const QSize &s, Qt::Alig
         return QPoint(0, 0);
     }
 
+    const QGraphicsItem *actualItem = item;
+
+    //its own view could be hidden, for instance if item is in an hidden Dialog
+    //try to position it using the parent applet as the item
+    if (!v->isVisible()) {
+        actualItem = item->parentItem();
+        if (!actualItem) {
+            const QGraphicsWidget *widget = qgraphicsitem_cast<const QGraphicsWidget*>(item);
+            if (widget) {
+                actualItem = qobject_cast<QGraphicsItem*>(widget->parent());
+            }
+        }
+
+        kDebug() << actualItem;
+
+        if (!v->isVisible() && actualItem) {
+            v = viewFor(actualItem);
+        }
+    }
+
+    if (!actualItem) {
+        actualItem = item;
+    }
+
+
     QPoint pos;
     QTransform sceneTransform = item->sceneTransform();
 
